src/renderer/api/create-terminal-api.injectable.ts

```diff
@@ -15,13 +15,13 @@ const createTerminalApiInjectable = getInjectable({
   instantiate: (di): CreateTerminalApi => {
     const hostedClusterId = di.inject(hostedClusterIdInjectable);
 
-    assert(hostedClusterId, "Can only create terminal APIs within a cluster frame");
+    return (query) => {
+      assert(hostedClusterId, "Can only create terminal APIs within a cluster frame");
```

Comment:
Having a falsy `hostedClusterId` cannot have magical property of also meaning that we are not within a cluster frame. For that, there needs to be something explicit.

Reply:
I think that an `assert` is pretty explicit. Plus this is something that will go away when we remove `iframe`'s.

Reply:
`assert` may be explicit, but `hostedClusterId` is not. Why not create a `weAreWithinClusterFrameInjectable` for clarity? Not that important right now, but let's not do something like this in future code. Resolved.